// q/q.js
/**
 * CSServerHandler.h
 * Copyright (c) 2015 NTT DATA. All rights reserved.
 * 
 */

 exports.callServiceWithURL = function (options) {
    //create the titanium HTTPClient
             console.log("authorized stirng ="+options.authorizedString);
             console.log("type ="+options.type);
             console.log("url ="+options.url);

    var httpClient = Titanium.Network.createHTTPClient();

    //set the httpclient's properties with the provided options
   
       
    //if and when response comes back
    //the success function is called
    httpClient.onload = function(e){
        options.success({
            data: httpClient.responseData, 
            text: httpClient.responseText,
            errorStatus: httpClient.status, 
            error: e.error
            
        });
        if(this.responseText){
                var jsonResponse = JSON.parse(this.responseText);
            }
    };
    
     httpClient.onerror = function(e){
        options.error({
            data: httpClient.responseData, 
            text: this.responseText,
            errorStatus: httpClient.status, 
            error: e.error
        });
        
        Titanium.API.log(" Error: " + this.status + " " + this.responseText + " " + e.error);

    };
        
    //open the connection
    
    httpClient.open(options.type, options.url);
    httpClient.setRequestHeader("Authorization",options.authorizedString);
    httpClient.setRequestHeader('Content-Type', "application/json");
    httpClient.setRequestHeader('Connection', 'close');
    httpClient.setRequestHeader("countryCode", "US");
    httpClient.setRequestHeader("source", "mobile");
    httpClient.setRequestHeader("partnerCode", "en");
    httpClient.validatesSecureCertificate = false;
    httpClient.timeout = 60000;
    httpClient.send(options.bodyString);
    
};
